### ISSUE TITLE
Amazon e-book/on-demand option

### ISSUE THREAD

**Issue by tonyskjellum** [mpi-forum/mpi-issues#508]
# Problem

Provide a print-on-demand book (two volumes) affordably from Amazon or equivalent.

# Proposal

Like OMP, provide on-demand print of MPI-4.1 standard and each tehreafter.

# Changes to the Text

Provide LaTeX multivolume version of the standard.

# Impact on Implementations

None.

# Impact on Users

Access to the print book is a plus.

# References and Pull Requests

N/A

**Comment by RolfRabenseifner** [comment]
As discussed, if a two-volumn solution will be done without font-size change, then it might be already an option for the MPI-4.0 pdf.

**Comment by mjklemm** [comment]
I have approached the editor of the OpenMP ARB. She will have a look at the current 4.0 document and tell us what needs to be fixed up. We can then have those fixes applied to the 4.0 document and later modify the 4.x document to automatically generate the right PDF file.

**Comment by RolfRabenseifner** [comment]
For spilitting MPI-4.0 into two volumes:
- Vol. I: pages i-xxii (Content, 22 pages) 
   xxiii-xl (Ackno. 18 pages), 1-632 (Chap1-12 [one-sided], 632 pages) 
   and 1065-1099 (Bibl.+Indexes 35 pages) = 707 pages
- Vol. II: pages i-xxii (Content, 22 pages) 
   633-1064 (Chap13 [Ext.Interfaces, I/O, Tool...]-19 [Lang.Bind.], Annex A+B, 432 pages) 
   and 1065-1099 (Bibl.+Indexes 35 pages) =  485 pages

I would do it such asymmetric to provide a maximum and most interestic chapters in Vol.1.
The common pages (i-xxii and 1065-1099) are needed in both volumes that all cross-referencing
with and without page numbers would work on paper.

The MPI-4.0 pdf format is 215.9 x 279.4 mm and its margins are about 22mm (left and right on pages with lin-numbers), 14mm (top, chapter pages), and 17mm (bottom, chapter titles). The used page size (i.e., without margins) is therefore ~ 172 x 248 mm 

OpenMP (711 pages) is printed with 203.2 x 254.0 (x 40.9) mm. The pdf has 190.5 x 228.6 mm. I.e., pdf may be enlarged for printing by a factor between about 1.067 (horizontal ratio) and 1.111 (vertical ratio). My book is in my real office. Therefore I cannot check the real factor from my home-office.

When MPI has to go with the same prining format 203.2 x 254.0 mm, then we have to shrink by a factor betwen 0.941 (horizontal) and 0.818 (vertical). This would imply used printed areas 
- between 162 x 233 (factor 0.941, only 30mm/10mm inner and outer margins, and 10mm margins at top and bottom), 
- and 141 x 203 (factor 0.818, 37mm/25mm inner and outer margins, and 25mm margins at top and bottom).

For MPI-3.1, we had 145 x 210 mm page size and 19mm/10mm inner and outer margins, and 11mm/12mm margins at top and bottom). This was enough with a hardcover sewn binding. For softcover, I would recommend a bit larger inner margin.
Therefore, the values for reduction with 0.941 should be perfect, provided that the margins of even and odd pages are treated in such different way.

The two volums would have 707 pages (40.7mm thick book, 1.70 kg) and 485 pages (27.9mm thick book, 1.17 kg) 
or in total 68.6mm thick and 2.87 kg. 

This is what I can recommend from my experience with publishing MPI-2.1, 2.2, 3.0 and 3.1.

**Comment by wgropp** [comment]
Thanks, Michael. I’m also looking at simplifying and modernizing the document design - that should make it easier to adapt to whatever requirements there are for these alternative versions.

Bill

William Gropp
Director, NCSA
Thomas M. Siebel Chair in Computer Science
University of Illinois Urbana-Champaign
IEEE-CS President-Elect

**Comment by wgropp** [comment]
Thanks for the information, Rolf.

Bill

William Gropp
Director, NCSA
Thomas M. Siebel Chair in Computer Science
University of Illinois Urbana-Champaign
IEEE-CS President-Elect

**Comment by mjklemm** [comment]
The page limits according to the OpenMP ARB's editor are as follows:

o	8 x 10”: 828 pp 
o	Letter: 590 pp
o	A4: 780 pp

So, the for 8x10 format, we'd be good with Rolf's proposal and how he split the document.

**Comment by RolfRabenseifner** [comment]
Then we have also the opportunity that we'll stay for longer with "Vol. I" within the 828 page limit.
It would be great if MPI-4.0 would be already printable on demand.

**Comment by mjklemm** [comment]
I'm on it. :-)

**Comment by jeffhammond** [comment]
What's the status of this?

**Comment by wgropp** [comment]
Thanks for bringing this up. The document updates that I've been pushing are the first step for this. I need to go and look at the print-on-demand requirements. I expect to need a way to split the standard into two volumes and manage the cross references.

**Comment by RolfRabenseifner** [comment]
I would recommend to split the pdf, i.e., to completely  keep the page numbers, i.e., to have a Vol. I with a gap from Chapters 13 until Annex B, and a Vol. II with a gap from Acknoledgments to Chap. 12. 
I would add a page + empty page into these gaps to tell that the missing pages are in the other volume.
I would use pdftk (under Linux) to substitute the parts by the special double page.

**Comment by mjklemm** [comment]
The main things to resolve:

- Reduce the page count to be below the maximum pages for KDP.  Rolf has a good proposal there and splitting can be done easily.
- Enlarge the pages so that the PDF has enough margin and bleed marks for the printing process.  I can have a look at this and see if pdftk can do that (I think so).
- Find someone who is fluent in KDP to create the book and publish it.  I can ask the OpenMP ARB staff to do that, but I cannot sensibly pay for that from the OpenMP ARB budget.  We'd need to find on the order 100-200 USD in labor to pay the contractor for that.

**Comment by wesbland** [comment]
Here's the help topic on the restrictions for KDP for [paperback](https://kdp.amazon.com/en_US/help/topic/G201834340) and [hardcover](https://kdp.amazon.com/en_US/help/topic/GHT976ZKSKUXBB6H). Basically you can't go over 828 pages for paperback and 550 pages for hardcover.

They also have a bunch of resources for how to format everything in [paperback](https://kdp.amazon.com/en_US/help/topic/G201834190) and [hardcover](https://kdp.amazon.com/en_US/help/topic/GKYZRXFBZH2LDXAK).

If we end up having an up-front cost that we need to resolve, that's something we'll have to figure out separately given that we are in the process of setting up our new financial system with [SPI](https://www.spi-inc.org).

**Comment by mjklemm** [comment]
KDP does not have costs associated with the publishing process.  The only cost incurred would be for a contractor, who knows the KDP process well and would be helpful in getting the KDP process sorted quickly.

If I find the time in the next two/three weeks, I can try to make an attempt to reformat the PDF without regenerating it from the LaTeX source, but by modifying the PDF directly.

**Comment by wgropp** [comment]
Thanks, @mjklemm . I'm ok if we do have to reformat from the LaTeX, though that will require more work.

**Comment by mjklemm** [comment]
The current trim size (textwidth x textheight) of the 4.0 document is 6x9.

The old book series had a trim size of about 4.5in x 7.0in.  That's a format we cannot do with KDP.  The smallest trim size we can do with KDP is 5x8.  In that format, we can do up to 828 pages.  We'd have to slightly scale the content down, but one of the dimensions will not exactly fit the KDP trim sizes (e.g. 5x8 will effectively be 5x7.5 when scaled from 6x9).  This is happens for any of the formats that are smaller than 6x9.

So, I guess it'll be better to stick to 6x9 and just make sure that we have enough space around the content to make KDP happy.  In the end, the final book will be a bit bigger than the previous MPI reports that Rolf was publishing.

Let me know which route you want me to pursue and I will start digging deeper into getting the format right.

**Comment by wgropp** [comment]
I think that we should stick with 6x9.  Thanks for looking into this.

**Comment by wesbland** [comment]
I just want to point out (probably unnecessarily) that we want to make sure that any changes can hopefully be replicated in the build system. Hopefully we publish MPI 4.1 in another year or so and I'd hate for you (or someone else) to have to go through all of this manual work again. 😄

**Comment by mjklemm** [comment]
> I just want to point out (probably unnecessarily) that we want to make sure that any changes can hopefully be replicated in the build system.

Yes, I'm actually also looking at this to make a proposal.  I might make these changes for in a special 4.0 branch and automate them as soon as possible.

**Comment by mjklemm** [comment]
@tonyskjellum @schulzm @isubsmith Any change to allocate a tiny amount of budget to work out the printing of the 4.0 standard?

**Comment by wesbland** [comment]
Do you want to bother with 4.0 given that we're targeting 4.1 for late next year?

**Comment by schulzm** [comment]
I would also suggest to revisit this in the light of MPI 4.1

**Comment by schulzm** [comment]
I added to the December agenda for discussion

**Comment by mjklemm** [comment]
> Do you want to bother with 4.0 given that we're targeting 4.1 for late next year?

Yes, because it is fairly easy and I have done a first successful test.  And it would keep the releases consistent that each version had a book (and I'd get both volumes)

**Comment by jeffhammond** [comment]
are the covers going to be colorful and distinct from other versions?

**Comment by mjklemm** [comment]
We can do that.  All I need is the color that we should use for the cover.  We could also invest a tiny bit of money and have a designer design some artwork.

**Comment by RolfRabenseifner** [comment]
The existing series was yellow - orange - green - blue, and my plan for 4.0 was (7 years ago) purple.
As retiered person, I'm now completely out of this discussion and I wish all the best. And if you start with 4.0, I would be happy to buy them, because I'm still active.
In my courses, I always promoted to use the books as a very cheap extension for the screen. In addition, books are very helpful for diving into a topic, reading the intros, looking at the general index for additional important locations about the topic.
